packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js

```diff
@@ -1,24 +1,94 @@
+import cuid from 'cuid'
+import { observer } from 'mobx-react'
 import { func } from 'prop-types'
-import React from 'react'
+import React, { useState } from 'react'
+
+function InteractionLayer ({ activeDrawingTask, svg }) {
+  const [ activeMark, setActiveMark ] = useState(null)
+  
+  function convertEvent (event) {
+    const type = event.type
+
+    const clientX = event.clientX
+    const clientY = event.clientY
+    const svgEventOffset = getEventOffset(clientX, clientY)
+
+    const svgCoordinateEvent = {
+      pointerId: event.pointerId,
+      type,
+      x: svgEventOffset.x,
+      y: svgEventOffset.y
+    }
+
+    return svgCoordinateEvent
+  }
+
+  function getEventOffset (x, y) {
+    const svgEvent = svg.createSVGPoint()
+    svgEvent.x = x
+    svgEvent.y = y
+    const svgEventOffset = svgEvent.matrixTransform(svg.getScreenCTM().inverse())
+
+    return svgEventOffset
+  }
+
+  function onPointerDown (event) {
+    const activeTool = activeDrawingTask.activeTool
+    const activeMark = activeTool.createMark({
+      id: cuid(),
+      toolIndex: activeDrawingTask.activeToolIndex
+    })
+    activeMark.setCoordinates(convertEvent(event))
+    setActiveMark(activeMark)
+  }
+  
+  function onPointerMove (event) {
+    if (activeMark) {
+      activeMark.setCoordinates(convertEvent(event))
+    }
+  }
+
+  function onPointerUp (event) {
+    if (activeMark) {
+      activeMark.setCoordinates(convertEvent(event))
+    }
+    setActiveMark(null)
+  }
 
-function InteractionLayer ({ onPointerDown, onPointerMove, onPointerUp }) {
   return (
-    <rect
-      id='InteractionLayer'
-      width='100%'
-      height='100%'
+    <g
       onPointerDown={onPointerDown}
       onPointerMove={onPointerMove}
       onPointerUp={onPointerUp}
-      fill='transparent'
-    />
+    >
+      <rect
+        id='InteractionLayer'
+        width='100%'
+        height='100%'
+        fill='transparent'
+      />
+      {activeDrawingTask &&
+        activeDrawingTask.tools.map( tool => {
+          const marksArray = Array.from(tool.marks.values())
+          return marksArray.map(mark => {
+
+            const MarkingComponent = observer(mark.toolComponent)
```

Comment:
Why is this being wrapped with an `observer`? Later for the editing functionality?

Reply:
So that marks move on the screen when you drag them and change the coordinates in their model.